FinSearchUnified/Tests/Bundle/SearchBundle/CriteriaRequestHandler/FindologicFacetCriteriaRequestHandlerTest.php

```diff
@@ -0,0 +1,305 @@
+        ];
+        $criteria = new Criteria();
+        $request = new RequestHttp();
+        $request->setModuleName('frontend');
+        $request->setParam('sSearch', 'text');
+        $request->setParams($parameter);
+        $productAttributeFacet = new ProductAttributeFacet($field, $mode, $formFieldName, $label);
+        $customFacet = new CustomFacet();
+        $customFacet->setFacet($productAttributeFacet);
+
+        // Create Mock object for Shopware Front Request
+        $front = $this->getMockBuilder(Front::class)
+            ->setMethods(['Request'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $front->expects($this->exactly(6))
+            ->method('Request')
+            ->willReturn($request);
+        $context = Shopware()->Container()->get('shopware_storefront.context_service')->getShopContext();
+        // Assign mocked session variable to application container
+        Shopware()->Container()->set('front', $front);
+
+        $sessionArray = [
+            ['isSearchPage', true],
+            ['findologicDI', false]
+        ];
+        // Create Mock object for Shopware Session
+        $session = $this->getMockBuilder(Session::class)->getMock();
+        $session->expects($this->atLeastOnce())
+            ->method('offsetGet')
+            ->willReturnMap($sessionArray);
+        // Assign mocked session variable to application container
+        Shopware()->Container()->set('session', $session);
+
+        // Create Mock object for Shopware Config
+        $config = $this->getMockBuilder(Config::class)
+            ->setMethods(['offsetGet', 'offsetExists'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $config->expects($this->atLeastOnce())
+            ->method('offsetGet')
+            ->willReturnMap($configArray);
+        // Assign mocked config variable to application container
+        Shopware()->Container()->set('config', $config);
+
+        $result = StaticHelper::useShopSearch();
+        $this->assertFalse($result, 'Expected FINDOLOGIC search to be triggered but it was not');
+
+        $customFacetServiceMock = $this->createMock(CustomFacetService::class);
+        $customFacetServiceMock->expects($this->once())->method('getList')->willReturn([$customFacet]);
+
+        $facetCriteriaRequestHandler = new FindologicFacetCriteriaRequestHandler($customFacetServiceMock);
+        $facetCriteriaRequestHandler->handleRequest($request, $criteria, $context);
+
+        $condition = $criteria->getConditions();
+        if ($hasCondition === false) {
+            $this->assertEmpty($condition);
+        }
+        if ($hasCondition === true) {
+            $this->assertNotEmpty($condition);
+            $this->assertTrue($criteria->hasCondition('product_attribute_' . $field));
+            $condition = $criteria->getCondition('product_attribute_' . $field);
+            $conditionField = $condition->getField();
```

Comment:
As stated in the story: "Passed criteria object has a condition of type ProductAttributeCondition"
Please assert that `FinSearchUnified/Bundle/SearchBundle/Condition/ProductAttributeCondition` is used, so we are sure not the default Shopware ProductAttributeCondition is used.